functional-tests/devscripts/run_tests.sh

```diff
@@ -91,75 +91,106 @@ echo "User name printed in format: 3 first letters, space, the rest of letters.
 
 #PR CHECK
 if [[ "$PR_CHECK_BUILD" == "true" ]]; then
-	HOST_URL=$(echo ${RH_CHE_AUTOMATION_SERVER_DEPLOYMENT_URL} | cut -d"/" -f 3)
-	echo "Running test against developer cluster. URL: $HOST_URL"
-	CHE_OSIO_AUTH_ENDPOINT="https://auth.prod-preview.openshift.io"
-	path="$(pwd)"
-	
-	mkdir report
-	
-	docker run \
-	   -v $path/report:/root/rh-che/local_tests/report:Z \
-	   -v $path/e2e-saas/:/root/rh-che/local_tests:Z \
-	   -e USERNAME=$RH_CHE_AUTOMATION_CHE_PREVIEW_USERNAME \
-	   -e PASSWORD=$RH_CHE_AUTOMATION_CHE_PREVIEW_PASSWORD \
-	   -e URL=http://$HOST_URL \
-	   --shm-size=256m \
-	quay.io/openshiftio/rhchestage-rh-che-e2e-tests
-	RESULT=$?
-	
-	mkdir -p ./rhche/${JOB_NAME}/${BUILD_NUMBER}/e2e_report
-	cp -r ./report ./rhche/${JOB_NAME}/${BUILD_NUMBER}/e2e_report
+  HOST_URL=$(echo ${RH_CHE_AUTOMATION_SERVER_DEPLOYMENT_URL} | cut -d"/" -f 3)
+  echo "Running test against developer cluster. URL: $HOST_URL"
+  CHE_OSIO_AUTH_ENDPOINT="https://auth.prod-preview.openshift.io"
+  path="$(pwd)"
+  
+  mkdir report
+
+  #check version in rh-che pom.xml to use correct version of tests.
+  version=$(getVersionFromPom)
+  if [[ -z "${version}" ]]; then
+    echo "[ERROR]: Could not find version in pom.xml."
+    exit 1
+  fi
+  rhche_image="quay.io/openshiftio/rhchestage-rh-che-e2e-tests:${version}"
+
+  docker pull $rhche_image
+  RESULT=$?
+
+  #reuse image if exists or build new image for test
+  if [[ $RESULT != 0 ]]; then
```

Comment:
I'm not sure... wouldn't it be nicer to just use the command in if?
```
if docker pull "$rhche_image"; then
```
? 
Your call ;-)